src/m365/entra/commands/group/group-set.spec.ts

```diff
@@ -0,0 +1,353 @@
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { displayName: 'Microsoft 365 Group', description: 'Microsoft 365 group', mailNickname: 'Microsoft365Group', visibility: 'Public', newDisplayName: '365 group' } });
+  });
+
+  it('successfully updates group with owners specified by ids', async () => {
+    sinon.stub(request, 'patch').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups/${groupId}` &&
+        JSON.stringify(opts.data) === JSON.stringify({
+          "description": 'Microsoft 365 group',
+          "mailNickName": 'Microsoft365Group',
+          "visibility": 'Public'
+        })) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch') {
+        return {
+          responses: Array(2).fill({
+            status: 204,
+            body: {}
+          })
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { id: groupId, description: 'Microsoft 365 group', mailNickname: 'Microsoft365Group', visibility: 'Public', ownerIds: userIds.join(',') } });
+    assert.deepStrictEqual(postStub.lastCall.args[0].data.requests, addOwnersRequest);
+  });
+
+  it('successfully updates group with members specified by user names', async () => {
+    sinon.stub(entraUser, 'getUserIdsByUpns').withArgs(userUpns).resolves(userIds);
+    sinon.stub(entraGroup, 'getGroupIdByDisplayName').withArgs('Microsoft 365 Group').resolves(groupId);
+
+    sinon.stub(request, 'patch').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups/${groupId}` &&
+        JSON.stringify(opts.data) === JSON.stringify({
+          "description": 'Microsoft 365 group',
+          "mailNickName": 'Microsoft365Group',
+          "visibility": 'Private'
+        })) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch') {
+        return {
+          responses: Array(2).fill({
+            status: 204,
+            body: {}
+          })
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { displayName: 'Microsoft 365 Group', description: 'Microsoft 365 group', mailNickname: 'Microsoft365Group', visibility: 'Private', memberUserNames: userUpns.join(','), verbose: true } });
+    assert.deepStrictEqual(postStub.lastCall.args[0].data.requests, addMembersRequest);
+  });
+
+  it('handles API error when adding users to a group', async () => {
+    sinon.stub(entraGroup, 'getGroupIdByDisplayName').withArgs('Microsoft 365 Group').resolves(groupId);
+
+    sinon.stub(request, 'patch').callsFake(async opts => {
```

Comment:
For this test, we don't really care what the URL of the API is. The only thing we want to test is that we correctly handle errors, no mather what the URL is. So just like your last test, let's just resolve or reject directly whatever URL is passed.